mithril-common/src/signable_builder/cardano_stake_distribution.rs

```diff
@@ -0,0 +1,171 @@
+use anyhow::anyhow;
+use std::sync::Arc;
+
+use async_trait::async_trait;
+
+use crate::{
+    crypto_helper::{MKTree, MKTreeNode},
+    entities::{Epoch, ProtocolMessage, ProtocolMessagePartKey, StakeDistribution},
+    signable_builder::SignableBuilder,
+    StdResult,
+};
+
+#[cfg(test)]
+use mockall::automock;
+
+/// Stake Distribution Retriever
+#[cfg_attr(test, automock)]
+#[async_trait]
+pub trait StakeDistributionRetriever: Send + Sync {
+    /// Retrieve the [StakeDistribution] for a given epoch
+    async fn retrieve(&self, epoch: Epoch) -> StdResult<Option<StakeDistribution>>;
+}
+
+/// A [CardanoStakeDistributionSignableBuilder] builder
+pub struct CardanoStakeDistributionSignableBuilder {
+    cardano_stake_distribution_retriever: Arc<dyn StakeDistributionRetriever>,
+}
+
+impl CardanoStakeDistributionSignableBuilder {
+    /// Constructor
+    pub fn new(cardano_stake_distribution_retriever: Arc<dyn StakeDistributionRetriever>) -> Self {
+        Self {
+            cardano_stake_distribution_retriever,
+        }
+    }
+
+    fn compute_merkle_tree(pools_with_stake: StakeDistribution) -> StdResult<MKTree> {
+        let leaves: Vec<MKTreeNode> = pools_with_stake
+            .iter()
+            .map(|(k, v)| MKTreeNode::new(format!("({},{})", k, v).as_bytes().to_vec()))
```

Comment:
IMO It'd be better to use a new type for representing an entry in the stake distribution and have it implement `Into<MKTreeNode>`

Reply:
Done ✅ 
https://github.com/input-output-hk/mithril/pull/1847/files#diff-89c15ec6d54d0deeb1bedf3e6cdb5adf592a46c64fe6a7e1c91a59076b606ef6R24